sale_stock_available_to_promise_release/models/product_product.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_next_replenishment_date(self):
+        self.ensure_one()
+        picking_type_in = self.env.ref("stock.picking_type_in")
```

Comment:
You should search for picking type with code incoming. And ideally take incoming moves where the destination location is a child of the warehouse view location if you compute that in the scope of a SO where a warehouse given.

Reply:
+1 for picking type = incoming